src/commands/highlights.rs

```diff
@@ -0,0 +1,88 @@
+use super::*;
+use crate::Arc;
+
+/// add a new highlight for the user
+#[command("add_highlight")]
+#[usage("!add_highlight <word, can have spaces>")]
+pub async fn add_highlight(ctx: &client::Context, msg: &Message, args: Args) -> CommandResult {
+    let args = args.message().trim().to_string();
+    if args.contains(" ") {
+        msg.reply_error(
+            &ctx,
+            "Highlight can't contain a space for implementation/performance reasons",
+        )
+        .await?;
+        return Ok(());
+    } else if args.len() == 0 {
+        msg.reply_error(&ctx, "You must provide a argument. ")
+            .await?;
+        return Ok(());
+    }
+
+    let db: Arc<Db> = ctx.get_db().await;
+    match db.set_highlight(msg.author.id, args.clone()).await {
+        Ok(_) => {
+            msg.reply_success(
+                &ctx,
+                format!("You will be notified whenever someone says {}", args),
+            )
+            .await?
+        }
+        Err(e) => {
+            msg.reply_error(&ctx, format!("The database returned a error: {}", e))
+                .await?
+        }
+    };
+
+    Ok(())
+}
+
+/// get all highlights for your user
+#[command("get_highlights")]
+#[usage("!get_highlights")]
+pub async fn get_highlights(ctx: &client::Context, msg: &Message) -> CommandResult {
+    let db: Arc<Db> = ctx.get_db().await;
+
+    let highlights = db.get_highlights().await?;
+
+    let mut s = String::new();
+    'outer: for i in highlights {
```

Comment:
you can `for (word, users) in highlights` to make stuff nicer

Reply:
It's gone anyways because of your other suggestion.